types/execution/execution.interfaces.ts

```diff
@@ -0,0 +1,12 @@
+export type FilenameTransformFunc = (filename: string) => string;
+
+export interface UnprocessedExecResult {
+    code: number;
+    okToCache: boolean;
+    filenameTransform: FilenameTransformFunc;
+    stdout: string;
+    stderr: string;
+    execTime: string;
+}
+
+export type TypicalExecutionFunc = (executable: string, args: string[], execOptions: object) => IUnprocessedExecResult;
```

Comment:
What is `IUnprocessedExecResult` ? I couldn't find it anywhere else in this PR and my IDE is giving it the red squigglies (technical term).

Reply:
Should lose the I, its the type above, but it should not be an interface, just a class. Refactoring mess.

Reply:
got it! I'm still such a noob at typescript I'm asking what-i-hope-are-sensible questions